File: SQLiteRepository/SQLiteQuery/CreateTableQuery.cs
using System;
using System.Collections.Generic;
using System.Data.SQLite;

namespace SQLiteRepository.SQLiteQuery
{
    public class CreateTableQuery: ISqliteQuery 
    { 
        public string TableName { get; set; }
        public string Path { get; set; }
        public Dictionary<string, string> Fields { get; set; }
        public string PrimaryKey { get; set; }
        public IQueryResult Execute()
        {
            return CreateTableSqlite.CreateTable(this);
        }
        public IQueryResult Execute(SQLiteConnection conn, SQLiteTransaction tr)
        {
            return CreateTableSqlite.CreateTable(conn, tr, this);
        }


    }
}

File: SQLiteAdapters/CreateTableQuery.vb
Imports System.Collections.Generic
Imports System.Data.SQLite


Public Class CreateTableQuery
    Inherits BaseQuery
   
    Public Overrides Function Execute() As IQueryResult
        Return CreateTableSqlite.CreateTable(Me)
    End Function
    Public Overrides Function Execute(conn As SQLiteConnection, tr As SQLiteTransaction) As IQueryResult
        Return CreateTableSqlite.CreateTable(conn, tr, Me)
    End Function


End Class